eng/Version.Details.xml

```diff
@@ -41,30 +41,30 @@
       <Uri>https://github.com/dotnet/corefx</Uri>
       <Sha>0f7f38c4fd323b26da10cce95f857f77f0f09b48</Sha>
     </Dependency>
-    <Dependency Name="System.Runtime.CompilerServices.Unsafe" Version="4.7.1" CoherentParentDependency="Microsoft.NETCore.App.Runtime.win-x64">
+    <Dependency Name="System.Runtime.CompilerServices.Unsafe" Version="4.7.0" CoherentParentDependency="Microsoft.NETCore.App.Runtime.win-x64">
```

Comment:
Why is this version going backward? Should not merge this PR until either what we're getting from corefx is fixed (is there an issue?) or we pin some version.

Reply:
This is almost certainly doing what it is designed to do (pick the version from the newest build). note that doesn't mean the newest **version**, just the version from the newest build.  I'm investigating.

Reply:
I figured it out. It's yet another reason why CPD is so complicated. So 4.7.1 disappeared from the graph. under core-setup because core-setup has no direct dependency on System.Runtime.CompilerServices.Unsafe.

So the previous release generated the 4.7.1 version of that assembly. It updated here properly because core-setup was referencing a version of corefx that produced that package. When we switched over for 3.1.4's build, we stopped producing a new S.R.CU as well as other out of band packages. That means that core-setup for 3.1.4 is not referencing **any** of the February build's corefx binaries any longer. Thus, the newest version in the graph was what it was before, which was 4.7.0.

You can either pin this here, or I can fix up core-setup to reference S.R.C.U at the shipping 4.7.1 version.

The other fix that we **could** make would be to change CPD so that it does not take into account builds. It only looks at direct dependencies. Currently you can do the following:

- A->CPD B
- B does not directly reference A, but references a dependency C from a build which also produced A.
- A resolves appropriately.

The advantage here is that aspnetcore-tooling for instance, doesn't need to reference every binary from extensions that aspnetcore may try to CPD. This means less bloated version details files in repos. Most of these dependencies that you would have to add are never actually used in this repo, just used downstream. This is also totally fine in cases where we do not do incremental servicing. The downside, of course, is that you can have this happen in incremental servicing cases.

I'm leaning towards **not** allowing build lookup in the CPD algorithm and fixing up all the repos. Of course, things like this make me also want to ban CPD altogether in 5.0 branches. It's insanely complicated.